Description: fix some compiler warnings
Author: A. Schulze
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: smf-spf-2.0.2/smf-spf.c
===================================================================
--- smf-spf-2.0.2.orig/smf-spf.c	2012-09-07 10:19:07.000000000 +0200
+++ smf-spf-2.0.2/smf-spf.c	2012-09-07 10:19:34.000000000 +0200
@@ -222,7 +222,7 @@
 
 static unsigned long hash_code(register const unsigned char *key) {
     register unsigned long hash = 0;
-    register size_t i, len = strlen(key);
+    register size_t i, len = strlen((char *)key);
 
     for (i = 0; i < len; i++) {
 	hash += key[i];
@@ -258,7 +258,7 @@
 }
 
 static SPF_result_t cache_get(const char *key) {
-    unsigned long hash = hash_code(key);
+    unsigned long hash = hash_code((unsigned char*) key);
     cache_item *it = cache[hash & hash_mask(HASH_POWER)];
     time_t curtime = time(NULL);
 
@@ -270,7 +270,7 @@
 }
 
 static void cache_put(const char *key, unsigned long ttl, SPF_result_t status) {
-    unsigned long hash = hash_code(key);
+    unsigned long hash = hash_code((unsigned char*) key);
     time_t curtime = time(NULL);
     cache_item *it, *parent = NULL;
 
